docs/proposals/volumes.md

```diff
@@ -0,0 +1,370 @@
+SELinux type `nfs_t`.  Setting `virt_use_nfs` on a system would allow containers to use *any* NFS
+mount without the kubelet having to relabel the files.  However, the semantics would be different
+from what might be expected, since SELinux will not provide any protection against containers from
+one pod using files in an NFS mount belonging to another pod.
+
+NFS is a case where the `chcon` approach isn't going to solve the whole problem.  Instead, the NFS
+volume plugin will have to handle calling `mount` with the `context` argument in order to gain
+cross-container protection from SELinux.
+
+Not all distributed file systems support SELinux uniformly.  Some file systems provide only the
+coarse-level control via an SELinux boolean to determine whether containers can use them.  The
+internal API should account for this.
+
+## Proposed Design
+
+Our proposed design should minimize code for handling ownership and SELinux required in:
+
+1.  volume plugins
+2.  the kubelet
+
+### API changes
+
+There are a couple factors to consider for the API:
+
+1.  Currently, it is possible to specify a different SELinux context for each container in a pod
+2.  There must be a way to specify a supplemental group that all containers in a pod are part of
+3.  There should be a uniform way to specify in the API that a volume requires ownership / label
+    management
+
+As stated in the assumptions of this proposal, we will not secure containers in a pod from one
+another.  This greatly simplifies the story for SELinux -- we can make the SELinux context uniform
+across the containers in a pod and remove it from the container level entirely.  The pod-level
+security proposal deals with this in detail.
+
+The pod-level supplemental group is necessary to capture groups that cut across all containers in a
+pod, which is the simplest way to guarantee that all arbitrary combinations of UID/GIDs in
+containers can share a volume.  The pod-level security context proposal also deals with this change
+in detail.
+
+Finally, we must have a clear way for users to indicate that Kubernetes should manage ownership and
+labeling of a volume, when it is possible.  For this, the `VolumeSource` should have a new field,
+`Manage`, and the volume `Builder` interface have two new methods, `RequiresOwnershipManagement()`
+and `RequiresLabelManagement()`.
+
+The `Builder` implementation for each type will be responsible for correctly indicating whether a
+particular volume requires ownership and label management:
+
+1.  The hostPath builder should always return `false` for both management types
+2.  The empty dir builder and builders for plugins derived from it should always return `true`
+3.  The builders for distributed file systems should return the correct values based on the `Manage`
+    field of the volume source and the SELinux support of that volume type.
+
+TODO: persistent volumes
```

Comment:
I will read this entire PR carefully to understand how it will work with PV and then provide feedback.